engine/overworld/special_warps.asm

```diff
@@ -2,59 +2,61 @@ SpecialWarpIn::
 	call LoadSpecialWarpData
 	predef LoadTilesetHeader
 	ld hl, wd732
-	bit 2, [hl] ; dungeon warp or fly warp?
+	bit 2, [hl] ; special warp
 	res 2, [hl]
-	jr z, .next
-; if dungeon warp or fly warp
+	jr z, .debugNewGameWarp
 	ld a, [wDestinationMap]
-	jr .next2
+	jr .next
+.debugNewGameWarp
+	bit BIT_DEBUG_MODE, [hl]
+	jr z, .setNewGameMatWarp ; apply to StartNewGameDebug only
+	call PrepareNewGameDebug
+.setNewGameMatWarp
+	; This is called by OakSpeech during StartNewGame and
+	; loads the first warp event for the specified map index.
+	; Without the instruction below, the game implicitly
+	; runs "ld a, VIRIDIAN_CITY".
```

Comment:
Huh? The game doesn't implicitly run things. Do you mean that without this line, `a` would have the same value as `VIRIDIAN_CITY` when it reaches the `ld [wLastMap], a` line? And if so, why is that being discussed? "If we omit part of the code, things will be different."

Reply:
Assuming it's because debug mode skips over this line, that should be more explicit. "Debug mode skips over this line and warps to VIRIDIAN_CITY instead, since a coincidentally has that value when it's stored in wLastMap."

Reply:
> Huh? The game doesn't implicitly run things.

Like in the other situation, my goal was to describe this in simple enough terms.

> Debug mode skips over this line and warps to VIRIDIAN_CITY instead

It warps to PALLET_TOWN, not VIRIDIAN_CITY. To avoid confusion, the comment will be removed.

Reply:
In general I don't think "without this line of code, such-and-such RAM's garbage value would have such-and-such meaning/effect", unless there's an unmodified in-game bug/glitch/feature which relies on that particular garbage value. (Like the comment describing Missingno's base data, in the middle of trainer party data.)